### PR TITLE
Update to Quarkus 3.17.0 and Camel 4.9.0

### PR DESCRIPTION
**Pull request by christophd**
# Changes

- Update to Quarkus 3.17.0 and Camel 4.9.0

/kind enhancement

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,11 +17,10 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
-        <camel.version>4.8.1</camel.version>
+        <camel.version>4.9.0</camel.version>
```
Comment (matzew) [inline pom.xml:20]: thanks. I will take a look soon

this will directly address the OIDC issues, or is there more to it?

Comment (christophd) [reply pom.xml:20]: yes, the OIDC support is included in 4.9.0. Last piece missing is to make this configurable on the pre-built images. I am going to add some source config via EnvVars similar to what we have with TLS support so you can enable/disable OIDC on integration source via just configuration. Will be a follow-up PR

**Comment by matzew** [comment]
quick sanity test:

```
--- PASS: TestIntegrationSourceWithSinkRef (32.48s)
    --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef (20.12s)
        --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Prerequisite (0.00s)
        --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Setup (0.00s)
            --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Setup/install_sink (14.11s)
        --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Requirement (0.00s)
            --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Requirement/install_integrationsource (0.01s)
            --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Requirement/integrationsource_goes_ready (3.01s)
        --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Assert (0.00s)
            --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Assert/integrationsource_as_event_source_delivers_events (3.00s)
        --- PASS: TestIntegrationSourceWithSinkRef/SendsEventsWithSinkRef/Teardown (0.00s)

```

**Review by matzew — APPROVED** [review]
/lgtm
/approve

existing bits work

**Pull request merged**